qa/L0_backend_python/logging/logging_test.py

```diff
@@ -0,0 +1,219 @@
+if USE_GRPC and USE_HTTP:
+    USE_GRPC = False
+assert USE_GRPC or USE_HTTP, "USE_GRPC or USE_HTTP must be non-zero"
+
+_trials = ["python"]
+
+_max_queue_delay_ms = 10000
+
+_deferred_exceptions_lock = threading.Lock()
+_deferred_exceptions = []
+
+
+class LogTest(tu.TestResultCollector):
+
+    def setUp(self):
+        # The helper client for setup will be GRPC for simplicity.
+        self.triton_client_ = grpcclient.InferenceServerClient(
+            f"{_tritonserver_ipaddr}:8001")
+        self.precreated_shm_regions_ = []
+        global _deferred_exceptions
+        _deferred_exceptions = []
+
+    def add_deferred_exception(self, ex):
+        global _deferred_exceptions
+        with _deferred_exceptions_lock:
+            _deferred_exceptions.append(ex)
+
+    def check_deferred_exception(self):
+        # Just raise one of the exceptions...
+        with _deferred_exceptions_lock:
+            if len(_deferred_exceptions) > 0:
+                raise _deferred_exceptions[0]
+
+    def check_response(self,
+                       trial,
+                       bs,
+                       thresholds,
+                       requested_outputs=("OUTPUT0", "OUTPUT1"),
+                       input_size=16,
+                       shm_region_names=None,
+                       precreated_shm_regions=None):
+        try:
+            start_ms = int(round(time.time() * 1000))
+
+            
+            tensor_shape = (bs, input_size)
+            iu.infer_exact(
+                self,
+                trial,
+                tensor_shape,
+                bs,
+                np.float32,
+                np.float32,
+                np.float32,
+                swap=False,
+                model_version=1,
+                outputs=requested_outputs,
+                use_http_json_tensors=False,
+                use_grpc=USE_GRPC,
+                use_http=USE_HTTP,
+                skip_request_id_check=True,
+                use_streaming=False,
+                shm_region_names=shm_region_names,
+                precreated_shm_regions=precreated_shm_regions,
+                use_system_shared_memory=False,
+                use_cuda_shared_memory=False)
+
+            end_ms = int(round(time.time() * 1000))
+
+            lt_ms = thresholds[0]
+            gt_ms = thresholds[1]
+            if lt_ms is not None:
+                self.assertTrue(
+                    (end_ms - start_ms) < lt_ms,
+                    "expected less than " + str(lt_ms) +
+                    "ms response time, got " + str(end_ms - start_ms) + " ms")
+            if gt_ms is not None:
+                self.assertTrue(
+                    (end_ms - start_ms) > gt_ms,
+                    "expected greater than " + str(gt_ms) +
+                    "ms response time, got " + str(end_ms - start_ms) + " ms")
+        except Exception as ex:
+            self.add_deferred_exception(ex)
+
+    def check_setup(self, model_name, preferred_batch_sizes,
```

Comment:
I don't think we need this. Remove.

Reply:
Minimizing to an identity function allowed me to cut down the bulk of this file.